src/main/java/org/fogbowcloud/manager/core/plugins/identity/ldap/LdapIdentityPlugin.java

```diff
@@ -0,0 +1,344 @@
+			SearchControls ctls = new SearchControls();
+			ctls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+			ctls.setReturningAttributes(new String[0]);
+			ctls.setReturningObjFlag(true);
+			NamingEnumeration enm = ctx.search(this.ldapBase, filter, new String[] { uid }, ctls);
+
+			String dn = null;
+
+			if (enm.hasMore()) {
+				SearchResult result = (SearchResult) enm.next();
+				dn = result.getNameInNamespace();
+				name = extractUserName(result);
+			}
+
+			if (dn == null || enm.hasMore()) {
+				// uid not found or not unique
+				throw new NamingException("Authentication failed");
+			}
+
+			// Bind with found DN and given password
+			ctx.addToEnvironment(Context.SECURITY_PRINCIPAL, dn);
+			ctx.addToEnvironment(Context.SECURITY_CREDENTIALS, password);
+			// Perform a lookup in order to force a bind operation with JNDI
+			ctx.lookup(dn);
+
+			enm.close();
+
+			return name;
+
+		} catch (Exception e) {
+			LOGGER.error("Error while authenticate " + uid + " - Error: " + e.getMessage());
+			throw e;
+		} finally {
+			ctx.close();
+		}
+
+	}
+
+	private String extractUserName(SearchResult result) {
+		String nameGroup[] = result.getName().split(",");
+		if (nameGroup != null && nameGroup.length > 0) {
+			String cnName[] = nameGroup[0].split("=");
+			if (cnName != null && cnName.length > 1) {
+				return cnName[1];
+			}
+		}
+		return null;
+	}
+
+	private String encryptPassword(String password) throws NoSuchAlgorithmException, UnsupportedEncodingException {
+		if (this.encryptType == null || this.encryptType.isEmpty()) {
+			return password;
+		}
+
+		MessageDigest algorithm = MessageDigest.getInstance(this.encryptType);
+		byte messageDigest[] = algorithm.digest(password.getBytes("UTF-8"));
+
+		StringBuilder hexString = new StringBuilder();
+		for (byte b : messageDigest) {
+			hexString.append(String.format("%02X", 0xFF & b));
+		}
+
+		return PASSWORD_ENCRYPTED.replaceAll(ENCRYPT_TYPE, this.encryptType).replaceAll(ENCRYPT_PASS, hexString.toString());
+
+	}
+
+	@Override
+	public boolean isValid(String accessId) {
+		try {
+			getToken(accessId);
+			return true;
+		} catch (Exception e) {
+			return false;
+		}
+	}
+
+	protected String createSignature(JSONObject json) throws IOException, GeneralSecurityException,
+			NoSuchAlgorithmException, InvalidKeyException, SignatureException, UnsupportedEncodingException {
+		RSAPrivateKey privateKey = null;
+		privateKey = RSAUtils.getPrivateKey(this.privateKeyPath);
+		String signature = RSAUtils.sign(privateKey, json.toString());
+		return signature;
+	}
+
+	protected boolean verifySign(String tokenMessage, String signature) {
+		RSAPublicKey publicKey = null;
+		try {
+			publicKey = RSAUtils.getPublicKey(this.publicKeyPath);
+			return RSAUtils.verify(publicKey, tokenMessage, signature);
+		} catch (Exception e) {
+			LOGGER.error("Erro while trying to validate sing of the token.", e);
+			throw new RuntimeException("Erro while trying to validate sing of the token.");
```

Comment:
Same as above.

Reply:
Done :). Check [this](https://github.com/fogbow/fogbow-manager-core/commit/78864ce8eefde35b7118f3caa28d205a7ff8ce8d) out.